cg/meta/deliver.py

```diff
@@ -29,6 +30,7 @@ def __init__(
         sample_tags: list[set[str]],
         project_base_path: Path,
         delivery_type: str,
+        fastq_file_service: FastqFileService,
```

Comment:
Is there a reason for requiring this as input? Might have missed something but can't we just set it as FastqFileService() by default?

Reply:
No, it is a poor design to couple dependencies like that. Even if it seems stupid now, the `FastqFileService` might grow in complexity, requiring more dependencies to be instantiated. As such, we want to keep the logic for instantiating it separate from this class to decouple them.